node/chainSimulator/chainSimulator.go

```diff
@@ -202,7 +203,34 @@ func (s *simulator) SetState(address string, state map[string]string) error {
 		return fmt.Errorf("cannot find a test node for the computed shard id, computed shard id: %d", shardID)
 	}
 
-	return testNode.SetState(addressBytes, state)
+	return testNode.SetKeyValueForAddress(addressBytes, state)
+}
+
+// SetStateMultiple will set state for multiple addresses
+func (s *simulator) SetStateMultiple(stateSlice []*dtos.AddressState) error {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+
+	addressConverter := s.nodes[core.MetachainShardId].GetCoreComponents().AddressPubKeyConverter()
+	for _, state := range stateSlice {
+		addressBytes, err := addressConverter.Decode(state.Address)
+		if err != nil {
+			return err
+		}
+
+		shardID := sharding.ComputeShardID(addressBytes, s.numOfShards)
+		if bytes.Equal(addressBytes, core.SystemAccountAddress) {
+			// for system account address use the provided shard ID
+			shardID = state.ShardID
```

Comment:
I think that, for the system account we should use all 3 shards.

Reply:
removed